src/app/main.rs

```diff
@@ -753,10 +760,19 @@ where
                         debug!("inbound canonical={:?}", canonical);
 
                         let dst = canonical
+                            .or_else(|| {
```

Comment:
This is the one thing in this PR I wasn't completely sure about --- should `l5d-override-dst` take priority over `l5d-canonical-dst`, or the other way around? 

In practice, I think we won't see both on the same inbound request, since `l5d-canonical-dst` is expected to only be set by outbound linkerd proxies, while `l5d-override-dst` is expected to be set only by clients that are _not_ linkerd proxies (as the outbound proxy will strip that header after using it). So, the ordering may not actually _matter_ all that much.

Reply:
Good question. i _think_ `canonical-dst` should take precedence, since it would only have been set by a calling proxy. If `l5d-dst-override` had also been set, they should in theory be identical (though the override could potentially be non-canonical? maybe?)

Reply:
@olix0r yup, that confirms what i was thinking! thanks.